query/sources/impl/src/main/java/org/codice/ddf/admin/sources/wfs/discover/GetWfsConfigsAction.java

```diff
@@ -10,43 +10,49 @@
  * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
  * is distributed along with this program and can be found at
  * <http://www.gnu.org/licenses/lgpl.html>.
- **/
-package org.codice.ddf.admin.sources.actions.persist;
+ */
+package org.codice.ddf.admin.sources.wfs.discover;
 
-import static org.codice.ddf.admin.sources.sample.SampleFields.SAMPLE_SOURCES_INFO_LIST;
+import static org.codice.ddf.admin.sources.commons.SourceActionCommons.getSourceConfigurations;
+import static org.codice.ddf.admin.sources.services.WfsServiceProperties.SERVICE_PROPS_TO_WFS_CONFIG;
+import static org.codice.ddf.admin.sources.services.WfsServiceProperties.WFS_FACTORY_PIDS;
 
 import java.util.List;
 
 import org.codice.ddf.admin.api.fields.Field;
-import org.codice.ddf.admin.api.fields.ListField;
 import org.codice.ddf.admin.common.actions.BaseAction;
 import org.codice.ddf.admin.common.fields.base.ListFieldImpl;
 import org.codice.ddf.admin.common.fields.common.PidField;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
 import org.codice.ddf.admin.sources.fields.SourceInfoField;
 
 import com.google.common.collect.ImmutableList;
 
-public class DeleteSource extends BaseAction<ListField<SourceInfoField>> {
+public class GetWfsConfigsAction extends BaseAction<ListFieldImpl<SourceInfoField>> {
 
-    public static final String NAME = "deleteSource";
+    public static final String ID = "wfsConfigs";
 
     public static final String DESCRIPTION =
-            "Delete's the given pid of a source. Returns back a list of all currently configured sources";
+            "Retrieves all currently configured WFS sources. If a source pid is specified, only that source configuration will be returned.";
 
     private PidField pid;
 
-    public DeleteSource() {
-        super(NAME, DESCRIPTION, new ListFieldImpl<>(SourceInfoField.class));
+    private ConfiguratorFactory configuratorFactory;
+
+    public GetWfsConfigsAction(ConfiguratorFactory configuratorFactory) {
+        super(ID, DESCRIPTION, new ListFieldImpl<>(SourceInfoField.class));
+        this.configuratorFactory = configuratorFactory;
         pid = new PidField();
     }
 
     @Override
-    public List<Field> getArguments() {
-        return ImmutableList.of(pid);
+    public ListFieldImpl<SourceInfoField> performAction() {
+        return getSourceConfigurations(WFS_FACTORY_PIDS,
```

Comment:
We should be validating that the pid exists before trying to retrieve the config. Make sure other identical actions are doing this as well.